operator_service/utils.py

```diff
@@ -82,13 +87,26 @@ def get_signer(signature, message):
 
 
 def process_provider_signature_validation(signature, original_msg, nonce):
-    if not signature or not original_msg:
-        return f"`providerSignature` of agreementId is required.", 400, None
-    original_msg = f"{original_msg}{nonce}"
     try:
         address = get_signer(signature, original_msg)
     except Exception as e:
         return "Failed to recover address", 400, None
+
+    db_nonce = get_nonce_for_certain_provider(address)
+
+    if db_nonce and float(nonce) < float(db_nonce):
```

Comment:
```suggestion
    if db_nonce and float(nonce) <= float(db_nonce):
```

Reply:
provider can use the same nonce every time and your condition is going to allow it